src/catalog/database_catalog.cpp

```diff
@@ -1083,6 +1088,88 @@ const IndexSchema &DatabaseCatalog::GetIndexSchema(transaction::TransactionConte
   return *reinterpret_cast<IndexSchema *>(ptr_pair.first);
 }
 
+std::vector<std::pair<common::ManagedPointer<storage::index::Index>, const IndexSchema &>>
+DatabaseCatalog::GetIndexObjects(terrier::transaction::TransactionContext *txn, terrier::catalog::table_oid_t table) {
+  // Step 1: Get all index oids on table
+  // Initialize PR for index scan
+  auto indexes_oid_pri = indexes_table_index_->GetProjectedRowInitializer();
+
+  // Do not need projection map when there is only one column
+  TERRIER_ASSERT(get_class_object_and_schema_pri_.ProjectedRowSize() >= indexes_oid_pri.ProjectedRowSize() &&
+                     get_class_object_and_schema_pri_.ProjectedRowSize() >= get_indexes_pri_.ProjectedRowSize() &&
+                     get_class_object_and_schema_pri_.ProjectedRowSize() >=
+                         classes_oid_index_->GetProjectedRowInitializer().ProjectedRowSize(),
+                 "Buffer must be allocated to fit largest PR");
+  auto *const buffer = common::AllocationUtil::AllocateAligned(get_class_object_and_schema_pri_.ProjectedRowSize());
+
+  // Find all entries for the given table using the index
+  auto *indexes_key_pr = indexes_oid_pri.InitializeRow(buffer);
+  *(reinterpret_cast<table_oid_t *>(indexes_key_pr->AccessForceNotNull(0))) = table;
+  std::vector<storage::TupleSlot> index_scan_results;
+  indexes_table_index_->ScanKey(*txn, *indexes_key_pr, &index_scan_results);
+
+  // If we found no indexes, return an empty list
+  if (index_scan_results.empty()) {
+    delete[] buffer;
+    return {};
+  }
+
+  std::vector<index_oid_t> index_oids;
```

Comment:
Stupid performance optimization: can we reserve the size of index_oids since we know it from the size of index_scan_results?